README.md

```diff
@@ -57,6 +57,84 @@ val sponsoredFundTransaction =
     wallet.fund(sourceAddress, destinationAddress, sponsorAddress = sourceAddress)
 ```
 
+### `Wallet.addAssetSupport()`
+
+```kotlin
+fun addAssetSupport(
+    sourceAddress: String,
+    assetCode: String,
+    assetIssuer: String,
+    trustLimit: String = "",
+    sponsorAddress: String = ""
+): Transaction
+```
+
+Example
+
+```kotlin
+val sourceAddress = "GAMQTINWD3YPP3GLTQZ4M6FKCCSRGROQLIIRVECIFC6VEGL5F64CND22"
+val assetIssuer = "GBBD47IF6LWK7P7MDEVSCWR7DPUWV3NY3DTQEVFL4NAT4AQH3ZLLFLA5"
+
+val transaction = wallet.addAssetSupport(sourceAddress, "USDC", assetIssuer)
+```
+
+### `Wallet.removeAssetSupport()`
+
+```kotlin
+fun removeAssetSupport(
+    sourceAddress: String,
+    assetCode: String,
+    assetIssuer: String
+): Transaction
+```
```

Comment:
Something I just thought about is why do we need to specify `sourceAddress` for all these functions? Can't we just use the `Wallet` object's `publicKey`? We could make `sourceAddress` optional for most functions and it would only be necessary if you wanted a different source address that the wallet's public key.

Reply:
I agree it's quite annoying to keep providing `sourceAddress`, but the `Wallet` doesn't have a mechanism to store the public key at the moment. We'll definitely need to make this better once we have an idea of how the public key will be accessed.

The `sourceAddress` does make sponsored transactions more clear, though.

Reply:
> the Wallet doesn't have a mechanism to store the public key at the moment

🤔 maybe we should think about this.

What if instead of having `Wallet.create()` return an `AccountKeypair`, the `Wallet` object stored a regular stellar SDK `Keypair`?

So creating a wallet could look like this:
```
Wallet(keypair=Keypair.fromSecret(<secret key>))
```

Or, if you're looking to create a new account:

```
Wallet()  // automatically creates a keypair using `Keypair.random()`
```

Then the `Wallet` class could expose `Wallet.publicKey` and `Wallet.secretKey`.

Reply:
:thinking: What if we create a new method `Wallet.setAccountAddress(publicKey: String)` (simple setter) that has to be called after `Wallet(...)` init. That way the logic could be like this:
1. For new sign-ups: backend has no public key -> Wallet() -> Wallet.create() -> save public key to backend -> Wallet.setAccountAddress(publicKey) -> ...
2. For sign-ins: get public key from backend -> Wallet() -> Wallet.setAccountAddress(publicKey) -> ...

I'd like to keep `Wallet.create()` as is because it would be a single place where keys are generated and a secret key is stored in Keystore right after. Nothing else in the code would "know" about it.

@lijamie98 what are your thoughts on this?

Reply:
I don't think it makes sense to have to use `Wallet.create()` and then `Wallet.setAccountAddress()` to save the public key to the same object that just generated it.

I think it would make more sense if there was no `Wallet.create()`, since its effectively equivalent to `Keypair.random()`. So it would look something like this:

```
kp = Keypair.random()
wallet = Wallet(accountAddress=kp.publicKey)
```
or
```
kp = Keypair.random()
wallet = Wallet()
wallet.setAccountAddress(kp.publicKey)
```

Reply:
@brunomuler, do we want to rethink what the` Wallet.create()` method does? If it just generates a keypair, maybe it could be moved to a helper method?

Reply:
This discussion doesn't have to block the PR, but lets keep talking about it. Personally I don't think it makes sense to create functions like `Wallet.create()`, which is already supported by our SDKs via `Keypair.random()`.